dissect/target/plugins/filesystem/yara.py

```diff
@@ -1,63 +1,184 @@
 
+log = logging.getLogger(__name__)
+
 YaraMatchRecord = TargetRecordDescriptor(
     "filesystem/yara/match",
     [
         ("path", "path"),
         ("digest", "digest"),
         ("string", "rule"),
         ("string[]", "tags"),
+        ("string", "namespace"),
     ],
 )
 
+DEFAULT_MAX_SCAN_SIZE = 10 * 1024 * 1024
+
 
 class YaraPlugin(Plugin):
     """Plugin to scan files against a local YARA rules file."""
 
-    DEFAULT_MAX_SIZE = 10 * 1024 * 1024
-
     def check_compatible(self) -> None:
-        pass
+        if not HAS_YARA:
+            raise UnsupportedPluginError("Please install 'yara-python' to use the yara plugin.")
 
-    @arg("--rule-files", "-r", type=Path, nargs="+", required=True, help="path to YARA rule file")
-    @arg("--scan-path", default="/", help="path to recursively scan")
-    @arg("--max-size", "-m", default=DEFAULT_MAX_SIZE, help="maximum file size in bytes to scan")
+    @arg("-r", "--rules", required=True, nargs="*", help="path(s) to YARA rule file(s) or folder(s)")
+    @arg("-p", "--path", default="/", help="path on target(s) to recursively scan")
+    @arg("-m", "--max-size", default=DEFAULT_MAX_SCAN_SIZE, help="maximum file size in bytes to scan")
+    @arg("-c", "--check", default=False, action="store_true", help="check if every YARA rule is valid")
     @export(record=YaraMatchRecord)
-    def yara(self, rule_files, scan_path="/", max_size=DEFAULT_MAX_SIZE):
-        """Scan files up to a given maximum size with a local YARA rule file.
+    def yara(
+        self,
+        rules: list[str | Path],
+        path: str = "/",
+        max_size: int = DEFAULT_MAX_SCAN_SIZE,
+        check: bool = False,
+    ) -> Iterator[YaraMatchRecord]:
+        """Scan files inside the target up to a given maximum size with YARA rule file(s).
 
-        Example:
-            target-query <TARGET> -f yara --rule-file /path/to/yara_sigs.rule
+        Args:
+            rules: ``list`` of strings or ``Path`` objects pointing to rule files to use.
+            path: ``string`` of absolute target path to scan.
+            max_size: Files larger than this size will not be scanned.
+            check: Check if provided rules are valid, only compiles valid rules.
+
+        Returns:
+            Iterator yields ``YaraMatchRecord``.
         """
 
-        rule_data = "\n".join([rule_file.read_text() for rule_file in rule_files])
+        compiled_rules = process_rules(rules, check)
+
+        if not rules:
+            self.target.log.error("No working rules found in '%s'", ",".join(rules))
+            return
 
-        rules = yara.compile(source=rule_data)
-        for _, _, files in self.target.fs.walk_ext(scan_path):
-            for file_entry in files:
-                path = self.target.fs.path(file_entry.path)
+        if hasattr(compiled_rules, "warnings") and (num_warns := len(compiled_rules.warnings)) > 0:
+            self.target.log.warning("Yara generated %s warnings while compiling rules", num_warns)
```

Comment:
```suggestion
            self.target.log.warning("YARA generated %s warnings while compiling rules", num_warns)
```
To be consistent with the rest.